students/GKim/lesson02/series.py

```diff
@@ -0,0 +1,66 @@
+
+def fibonacci(n):
+    """ compute the nth Fibonacci number """
+    a, b = 0, 1
+    for _ in range(n-1):
+        a, b = b, a + b
+    return b
+    
+
+
+def lucas(n):
+    """ compute the nth Lucas number """
+    a, b = 2, 1
+    if n == 0:
+        return a
+    elif n == 1:
+        return b
+    else:
+        for _ in range(n-1):
+            a, b = b, a + b
+        return b
+
+
+def sum_series(n, n0=None, n1=None):
+    """
+    compute the nth value of a summation series.
+
+    :param n0=0: value of zeroth element in the series
+    :param n1=1: value of first element in the series
+    
+    This function should generalize the fibonacci() and the lucas(),
+    so that this function works for any first two numbers for a sum series.
+    Once generalized that way, sum_series(n, 0, 1) should be equivalent to fibonacci(n).
+    And sum_series(n, 2, 1) should be equivalent to lucas(n).
+    """
+    if n0 == None and n1 == None:
+        return fibonacci(n)
+
+    elif n0 == 2 and n1 == 1:
+        return lucas(n)
+
```

Comment:
This is very close! Remember that you will also have to handle series that don't start with 2,1 or no optional parameters. If n0 = 5 and n1 = 2 (or almost any other permutation), this sum_series won't do anything.

Hint: Your fibonacci and lucas functions are extremely similar. Think about how you can use one general function to do both of their jobs

Reply:
This looks good, nice job!